scripts/CodeCoverage.py

```diff
@@ -0,0 +1,189 @@
+
+def get_project_name_from_CMakeLists_file(full_file_path):
+    for line in get_file_content(filename=full_file_path):
+        found, project_name = get_library_to_build_name(line_str=line)
+        if found:
+            return True, project_name
+    return False, ""
+
+def print_error_and_usage(argParser, error):
+    print "Error:  " + error + "\n"
+    print argParser.print_help()
+    sys.exit(2)
+
+def santize_input_args(arg_parser, args):
+    if len(sys.argv) > 3:
+        print_error_and_usage(argParser=arg_parser, error="Too many arguments supplied.")
+
+def destroy_directory(dir_path):
+    shutil.rmtree(dir_path)
+
+def clean_and_build_directory(dir_path):
+    if os.path.exists(dir_path):
+        destroy_directory(dir_path=dir_path)
+    os.makedirs(dir_path)
+
+def remove_file_extension(filename):
+    return os.path.splitext(filename)[0]
+
+def strip_file_from_full_path(full_file_path):
+    return ntpath.basename(full_file_path)
+
+def unzip_file(full_file_path, directory_to_extract_to):
+    to_unzip = zipfile.ZipFile(full_file_path, 'r')
+    unzipped_dir_name = remove_file_extension(filename=strip_file_from_full_path(full_file_path=full_file_path))
+    full_unzipped_dir_path = directory_to_extract_to + "/" + unzipped_dir_name
+    if os.path.exists(full_unzipped_dir_path):
+        destroy_directory(full_unzipped_dir_path)
+    to_unzip.extractall(directory_to_extract_to)
+    to_unzip.close()
+
+
+def run_CMake_cmd():
+    CMAKE = '/usr/local/probe/bin/cmake'
+    DEBUG_FLAG = '-DUSE_LR_DEBUG=ON'
+    VERSION_FLAG = '-DVERSION=' + str(1)
+    ARG1_FLAGS = '-DCMAKE_CXX_COMPILER_ARG1:STRING=\'-Wall -Werror -g -gdwarf-2 -fno-elide-constructors -fprofile-arcs -ftest-coverage -O0 -fPIC -m64 -Wl,-rpath -Wl,. -Wl,-rpath -Wl,/usr/local/probe/lib -Wl,-rpath -Wl,/usr/local/probe/lib64 -fno-inline -fno-inline-small-functions -fno-default-inline\''
+    COMPILER_EXE_FLAG = '-DCMAKE_CXX_COMPILER=/usr/local/probe/bin/g++'
+    CMAKELISTS_DIR = '..'
+    CMAKE_STR = CMAKE + ' ' + DEBUG_FLAG + ' ' + VERSION_FLAG + ' ' + ARG1_FLAGS + ' ' + COMPILER_EXE_FLAG + ' ' + CMAKELISTS_DIR
+    CMAKE_CMD = [CMAKE_STR]
+    ret_code = subprocess.check_call(CMAKE_CMD, stderr=subprocess.STDOUT, shell=True)
+    print "CMake return code: " + str(ret_code)
+
+def compile_project():
+    COMPILE_CMD = ['make -j']
+    ret_code = subprocess.check_call(COMPILE_CMD, stderr=subprocess.STDOUT, shell=True)
+    print "Compile return code: " + str(ret_code)
+
+def run_UnitTestRunner(launch_dir):
+    CP_RUNNER_SCRIPT = 'cp ' + launch_dir + '/scripts/unitTestRunner.sh ' + launch_dir + '/build'
+    CP_RUNNER_SCRIPT_CMD = [CP_RUNNER_SCRIPT]
+    ret_code = subprocess.check_call(CP_RUNNER_SCRIPT_CMD, stderr=subprocess.STDOUT, shell=True)
+    print "Copy script return code: " + str(ret_code)
+    RUNNER_SCRIPT_CMD = ['sh unitTestRunner.sh']
+    try:
+        ret_code = subprocess.check_call(RUNNER_SCRIPT_CMD, stderr=subprocess.STDOUT, shell=True)
+    except:
```

Comment:
You can be specific about your exceptions, check into what you expect it to through.

```
except Exception as e:
    print("exception bro: {0}".format(e))
```